pkg/controller/pxc/controller.go

```diff
@@ -419,35 +402,33 @@ func (r *ReconcilePerconaXtraDBCluster) Reconcile(request reconcile.Request) (re
 		}
 		if len(o.Spec.HAProxy.ReplicasServiceType) > 0 {
 			//Upgrading service only if something is changed
-			if currentServiceReplicas.Spec.Type != o.Spec.HAProxy.ReplicasServiceType {
-				currentServiceReplicas.Spec.Ports = replicaPorts
-				currentServiceReplicas.Spec.Type = o.Spec.HAProxy.ReplicasServiceType
+			if haProxyServiceReplicas.Spec.Type != o.Spec.HAProxy.ReplicasServiceType {
+				haProxyServiceReplicas.Spec.Ports = replicaPorts
+				haProxyServiceReplicas.Spec.Type = o.Spec.HAProxy.ReplicasServiceType
 			}
 			//Checking default ServiceType
-		} else if currentServiceReplicas.Spec.Type != corev1.ServiceTypeClusterIP {
-			currentServiceReplicas.Spec.Ports = replicaPorts
-			currentServiceReplicas.Spec.Type = corev1.ServiceTypeClusterIP
+		} else if haProxyServiceReplicas.Spec.Type != corev1.ServiceTypeClusterIP {
+			haProxyServiceReplicas.Spec.Ports = replicaPorts
+			haProxyServiceReplicas.Spec.Type = corev1.ServiceTypeClusterIP
 		}
```

Comment:
now it can be simplified, it had no sense in obsolite magic
````
haProxyServiceReplicas.Spec.Ports = replicaPorts
if len(o.Spec.HAProxy.ReplicasServiceType) > 0 {
    haProxyServiceReplicas.Spec.Type = o.Spec.HAProxy.ReplicasServiceType
} else {
    haProxyServiceReplicas.Spec.Type = corev1.ServiceTypeClusterIP
}

Reply:
pushed this in my last changes)
f268026

Reply:
code is still here